---
layout: post
title: GCD 2014 Meeting overview
published: true
tags: science
categories: science
date: 2014-09-21 00:19:00
---

I spent most of last week at the [Genomics of Common Diseases](http://www.nature.com/natureconferences/gcd2014/index.html) meeting held by Nature Genetics and the Wellcome Trust. Below I summarize some of the big themes of the conference, and give the big takeaway message I got from each of the talks.

## Big themes

Before the meeting, Orli Bahcall presented a list of "big questions" regarding the state and future of genomics of common diseases. While going through the questions and absorbing the talks, I came up with the following big themes:

#### A need for better infrasctructure for genomics data sharing
- huge datasets, infrastructure will be huge (Chin, Altshuler,
Manolio, big projects like GTEx, 1000G)

#### Genome Editing
- genome editing (not discussed as much as I thought it would be, but
key) Zhang, Musunuru. my qusetions: different cellt ypes, how get it
to scale, need high TP functional assays

#### Challenges in discovery and interpretation of susceptibility loci
- partition h2 (hilary)
- better phentoyping (Tobin, Gregerson, Cho)
- need to better integrate data to interpret GWAS hits. hilary
but also think clever
computational methods to do this will be key (Hoskins,
Dongwon's poster on DNAse, Cox)
mention hoskins
overlap with qtls (Huyghe)
better null models (goldstein de novos, mt. model) and understand mutations (Gil, Molly)
figure out architecture (vineeta)

#### Challenges in bringing genomics to the clinic
- issue of penetrance. rarely see fully penetrant variant in complex
disease. crucial to also look at individuals *without*
the phenotype which have that variant.  (deCODE talk is key, Altshuler
global alliance)
- health care infrastructure
- infrastructure, Altshuler, UK 100K (hubbard)
- howard jacob: future of clinical sequencing (apple watch joke)

## Big messages from each talk
Note a couple are missing, mostly because some were closed to social media, but a couple of them I wasn't in the room for, sorry!

* **Bert Vogelstein (keynote)**
Deaths from cardiovascular disease over the last 60 years have
decreased about 70%, while deaths from cancer have stayed
stagnant. Why? Vogelstein suggests this is because most funding for
cancer research has focused on curing advanced cancers, but the most
promise for impact comes from *prevention* and *early detection*.

* **Lynda Chin**
We are generating overwhelmingly large cancer genomics datasets: whole genome
sequencing, transcriptomics, proteomics, and scores of epigenetic
marks. We need to make better visualization and data resources to
crowdsource analyses of these data to maximize its utility.

* **Jason Hoskins**
Hoskins tackles the important task of following up a GWAS hit for predisposition to pancreatic
cancer all the way to the causal variant and mechanism. He finds a
double insertion event tagged by the GWAS hit that forms a novel
binding site for LEF1 that silences DIS3 through long range interaction.

* **Brad Bernstein**
Berstein uses high throughput and single cell genomics to find the
factors driving stemness in glioblastomas.

* **Martin Tobin**
Lung function and COPD are highly heritable conditions, and
associations found by GWAS explain only a small portion of the genetic
variance. By smartly stratifying cohorts (here non-smokers
vs. smokers) we can gain better insight into variants contributing to
the disease.

* **Peter Gregersen**
"Phenotype. phenotype. phenotype." Focusing on rheumatoid arthritis, Gregersen emphasizes that often
disease phenotypes can be very heterogenous. We can increase our power
and effect sizes by looking at more homogenous phenotypes, such as
molecule "endophenotypes" underlying the disease.

* **Judy Cho**
Cho also emphasizes, this time in IBD, that a key to GWAS success is
identifying the correct molecular endophenotype. She identifies links
between IBD and mycobacterial susceptibility.

* **Eric Green (keynote)**
Eric Green expresses optimism about the future of genomic medicine,
and gives his 6 hottest areas upcoming areas of importance: (1) cancer
genomics, (2) pharmacogenomics, (3) genomic medicine "test drive"
programs, (4) prenatal and newborn genomic analysis, (5) clinical
genomics information systems, and (6) ultra-rare disease genomics.

* **Matthew Maurano**
Deep sequencing of DNaseI hypersensitivity sites and other molecular
phenotypes elucidates widespread allelic imbalance and shows cell type
specific differences in the genome's regulatory architecture.

* **Nancy Cox**
In order for us to accurately translate a variant into patient care,
we will need to integrate data from a large number of sources, which
by themselves have no clinical utility. Cox also presents a clever
approach of using genotypes to predict gene expression, then
associated expression levels directly with disease phenotypes to gain
insight into key genes related to disease.

* **Simona Volpi**
The GTEx project is generating large datasets of genotype and
expression data across many individuals in an large range of tissue
types that will serve as a valuable resource for interpreting the
effect of genetic variants.

* **Jeroen Huyghe**
Integrating GWAS hits with functional data will be a key to
interpretation. GWAS SNPs from Type 2 Diabetes alter expression in
relevant tissues of liver, pancreas, and skeletal muscle.

* **Howard Jacob**
"The future of common disease: phenotype with your apple watch, plus
sequencing in the clinic". Jacob claims the future of clinical
sequencing lies in whole genome, rather than whole exome sequencing,
which should capture a larger range of relevant variation.

* **Hreinn Stefannson (deCODE)**
Most association studies study the effect of a variant on individuals
*with a certain phenotype*. However, this does not allow us to answer
the question of penetrance. Here deCODE looks at how CNVs implicated in
schizophrenia and autism behave in normal control samples. They find
that these CNVs significantly affect cognitive function in controls,
separate from the manifest disease.

* **David Goldstein**
*De novo* mutations are in clear excess in patients arriving at the
genetics clinic. To evaluate the significance of these, it is
important to have an accurate null model of *de novo* mutation.

* **Leslie Biesecker**
A key objection to using sequencing results in the clinic is that the
penetrance of most variants is unknown. Currently, reported variants
suffer from a high false positive rate. Biesecker claims that we have
been underestimating the prior probability of disease, and that by
more accurately estimating this we may be able to boost the positive
predictive value of mutations returned by these tests.

* **David Hinds (23andMe)**
23andMe have collected a large genetic dataset from a diverse set of
Americans. Through online surveys, they achieve striking GWAS results
for such traits as "do you cry easily" and "are you a morning person".

* **Yaniv Erlich**
Using an awesome analysis of a 43 million person pedigree obtained
using social media, Erlich showed that the overwhelmingly best model
for the genetic architecture of longevity is an additive
model.

* **Vinneta Agarwala**
Are "synthetic associations", or rare variants on common haplotypes,
driving GWAS signals? Agarwala finds no evidence of this at the 9p21 locus implicated
in Type 2 Diabetes, suggesting the signal is due primarily to common
variants.

* **Hilary Finucane**
Using a novel statistical genetics metric, the LD score, we can more
accurately measure the heritability of polygenic traits, and can
partition heritability between different functional annotations.

* **Teri Manolio**
Manolio describes growing genomic medicine initiatives from around the
world. 

* **Tim Hubbard**
UK is sequencing 100,000 clinical grade genomes with a focus on rare
inherited diseases, cancer, and pathogens. This project will be
directly integrated with the health care system, and aims to develop
important infrastructure for integrating sequencing into health care.

* **David Altshuler**
To fully understand the genomics of many diseases, we will need to
look across hundreds of thousands or even millions of genomes. No
single institute has the capacity for this, and a key element of
progress will be to ensure interoperability of genomics data across
many institutions worldwide. Unless we take action, genomics data will
become "balkanized" in different silos. The Global Alliance seeks to
advance interoperability of genomics data by developing important ethical and
technical frameworks.

* **Steve Munger**
Many mouse studies are done using the Black 6 strain, and are found
not to recapitulate human responses. However, this
strain behaves like an outlier strain in many respects. At Jackson
Laboratory, they are creating an outbred mouse strain from 8 founder
strains to allow more complex studies of genotype-phenotype
relationships in mice.

* **Feng Zhang**
CRISPR-Cas9 allows precise genome-editing. It can be used for a huge
variety of genomics applications, including generating genome-scale
knockout libraries and editing specific genomic variants. Zhang's
group developed a cre-dependent Cas9 mouse, allowing cell-type
specific Cas9 expression and genome editing.

* **Kiran Musunuru**
In associated genotype to phenotype, we need to address four key
questions: (1) what is the causal variant? (2) How do we link the
causal variant to a gene? (3) What is the causal gene? (4) How do we
relate the causal gene to the phenotype? Musunuru is following up a
GWAS hit for Type 2 Diabetes and HDL in KLF14 using a CRISPR screen to
cover all of the variants at this locus.

* **Molly Przeworksi (keynote)**
There are a lot of basic properties of the human mutation process that
we still don't understand. Recent studies have estimated the germline
mutation rate using different methods, including looking at disease
related genes, phylogenetic relationships, and deep sequencing
trios, each of which gives different estimates. Przeworksi presents a
model accounting for mutation repair rates, paternal age, and
generating time and shows how it could reconcile estimates from
different methods.

* **Gil McVean**
Many germline mutations are not simple point mutations. How can we
access the full spectrum of de novo mutations? McVean looks at
plasmodium genomes using an assembly based approach to study all types
of variation in a hypothesis free way.

* **Joanna Mountain (23andme)**
23andme looks at the genetic landscape of individuals across the
US. Using data from its large user-base, they can infer different
ancestry components in diverse populations including Latinos, African
Americans, and Americans of European descent.

* **Rebecca Rothwell**
Rothwell develops a method of mitochondrial transmission in
humans. She shows that a "nucleoid model", in which groups of 5-10
mitochondria replicate as a unit, fits best to current data.

* **Julie Segre**
Segre shows the use of PacBio and Illumina sequencing technologies to
track outbreaks of hospital acquired infections.

